### PR TITLE
sending events for each parent event id of group

### PR DESCRIPTION
**Pull request by Topru333**

### CODE REVIEW

File: src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt [hunk 1]
```diff
@@ -44,6 +41,16 @@ val MessageGroup.parentEventId: String?
         }
         .firstOrNull { it != null }
 
+val MessageGroup.allParentEventIds: Set<String>
+    get() = messagesList.asSequence()
+        .map {
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt:46]:
```suggestion
        .mapNotNull {
```

File: src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt [hunk 2]
```diff
@@ -44,6 +41,16 @@ val MessageGroup.parentEventId: String?
...
+            when {
+                it.hasMessage() -> it.message.parentEventId.id.ifEmpty { null }
+                it.hasRawMessage() -> it.rawMessage.parentEventId.id.ifEmpty { null }
+                else -> null
+            }
+        }.filterNotNull().toSet()
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt:52]:
```suggestion
        }.toSet()
```

**Review by ivandruzhinin — APPROVED** [review]

File: src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt (hunk 2 above)
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt:52]:
```suggestion
    get() = messagesList.mapNotNullTo(HashSet()) {
        when {
            it.hasMessage() -> it.message.parentEventId.id.ifEmpty { null }
            it.hasRawMessage() -> it.rawMessage.parentEventId.id.ifEmpty { null }
            else -> null
        }
    }
```

File: src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt [hunk 3]
```diff
@@ -97,9 +94,5 @@ fun MessageGroup.toErrorMessageGroup(exception: Throwable, protocol: String) : M
 }
 
 fun RawMessage.toMessageMetadataBuilder(protocol: String): MessageMetadata.Builder {
-    return MessageMetadata.newBuilder()
-        .setId(metadata.id)
-        .setTimestamp(metadata.timestamp)
-        .setProtocol(protocol)
-        .putAllProperties(metadata.propertiesMap)
+    return MessageMetadata.newBuilder().setId(metadata.id).setTimestamp(metadata.timestamp).setProtocol(protocol).putAllProperties(metadata.propertiesMap)
```
Comment (ivandruzhinin) [inline src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt:97]: Why?

Comment (Topru333) [reply src/main/kotlin/com/exactpro/th2/codec/util/MessageUtil.kt:97]: sorry trying to setup auto reformat

**Review by ivandruzhinin — APPROVED** [review]

**Pull request merged**